serverless/src/shared/buildHierarchicalCsvPaths.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @file buildHierarchicalCsvPaths.js
+ * @description Builds hierarchical CSV paths for a given scheme and concept map.
+ */
+
+import { cloneDeep } from 'lodash'
+
+import { formatCsvPath } from '@/shared/formatCsvPath'
+import { getNarrowers } from '@/shared/getNarrowers'
+import { isCsvLongNameFlag } from '@/shared/isCsvLongNameFlag'
+import { isCsvProviderUrlFlag } from '@/shared/isCsvProviderUrlFlag'
+
+/**
+ * Builds hierarchical CSV paths recursively.
+ *
+ * @param {number} csvHeadersCount - The number of CSV headers.
+ * @param {Object} providerUrlsMap - Map of provider URLs.
+ * @param {Object} longNamesMap - Map of long names.
+ * @param {Object} scheme - The scheme object.
+ * @param {Object} n - The current node in the hierarchy.
+ * @param {Object} map - The concept map.
+ * @param {Array} path - The current path (default: []).
+ * @param {Array} paths - The array to store all paths (default: []).
+ * @returns {Promise<void>}
+ */
+// eslint-disable-next-line max-len
+export const buildHierarchicalCsvPaths = async (csvHeadersCount, providerUrlsMap, longNamesMap, scheme, n, map, path = [], paths = []) => {
```

Comment:
anytime you have a large list of parameters, use {} for them instead.  i.e, ({ csvHeadersCount, providerUrlsMap, longNamesMap, scheme, n, map, path = [], paths = [] })

Reply:
Done